doc/plugin-criteria.md

```diff
@@ -0,0 +1,51 @@
+Krew plugin criteria
+===
+
+Rationale
+---
+
+Krew is a `kubectl` plugin manager.
+
+A plugin provides a technical solution to at least one _task_ concerning a problem around kubernetes.
+As plugins are invoked via `kubectl <plugin-name>`, they should feel like natural extensions of the `kubectl` command in a way that some kubectl task is extended.
+Stated reversely: without the plugin, users would typically try to solve the plugin task mainly with kubectl.
+
+A plugin should be classified in the following categories:
+
+- Purpose
+- Category
+- Usability
+
+Purpose
+---
+
+This task should be classified in one of the categories:
+
+| #  | Classification                                                                                                                                                                                                                 | Recommendation                                                                                                                                                                                            |
+|----|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1. |  The task is already solved by kubectl.                                                                                                                                                                                        |  Mere command wrappers are not acceptable, as those are better covered by shell aliases.  However, convenience and productivity improvements are welcome (examples: `konfig`, `change-ns`, `view-secret`) |
+| 2. | The task provides a different solution to a problem which is also solved by kubectl. However, the task would be impossible or very hard to reproduce with kubectl alone. Examples: `access-matrix`, `get-all`, `mtail`, `tail` |  In general welcome unless it overlaps significantly with other plugins or has a bad usability.                                                                                                           |
+| 3. | The task addresses a completely different problem than any of the kubectl subcommands, but is clearly focused on k8s.                                                                                                          |  Gray area. If the task fits well into `kubectl`, it may be accepted (for example `kubectl debug`).                                                                                                       |
```

Comment:
The example of `kubectl debug` would appear to be against the [naming guidelines](https://github.com/kubernetes-sigs/krew/blob/master/docs/NAMING_GUIDE.md), in particular "Be Specific". It feels like the Login example provided there.

Reply:
+1 we rejected several plugins named `debug`.

A common task they do is: `kubectl cp` a busybox binary, `kubectl exec` to install busybox symlinks, then `kubectl exec -- sh` to drop user to a shell inside the container. Although I think we accepted one named `pod-debug`.

Reply:
Oops. I had https://github.com/kubernetes-sigs/krew-index/pull/135 in mind, but forgot that the agreement was to change the name.
Changed to `kubectl debug-pod`.